Task/hchow/tlt 1870 update selenium tests to work in new project

hktest opened the pull request:
- Added in the new dashboard page object for the new canvas_course_admin_tools project
- Added in a new course_admin_base_test_case
- Modified existing existing base_test_case for Manage People tool (given the new dashboard view)

No merge conflicts, but @elliottyates branch has downstream changes that I will need to pull (rebase?) before I merge into his branch.

hktest committed:
- Created new page object for Manage Course dashboard
- Cleaned up code to point to new project
- Refactored some code and fixed some PEP8 issues.

hktest committed:
- Fixed up manage_people base_test_case to inherit from CourseAdminBaseTestCase (new dashboard view)
- Cleaned up code, pep8 issues

elliottyates commented:
@elliottyates

elliottyates commented on canvas_course_admin_tools/settings/local.py:
```diff
@@ -42,6 +42,9 @@
         'base_path': 'api/course/v2'
     },
+    'manage_course': {
+        'relative_url': 'courses/27/external_tools/170',  # dev (Manage Course)
```
Does this mean we can remove the URLs for `manage_people`, and that we need to move the comment above those URLs up here?

hktest replied:
Yup, that's the plan!  (And I remembered it from your last PR feedback for the canvas_account_admintools_tool).  :)   I haven't done it yet because the Manage People is currently down for me (note scrumteam chat), so I'm not able to run any of the existing manage people tests to make sure they all run as they should. The plan is to remove the manage people URL once I can successfully run them since the dashboard click should take care of the need to explicitly provide the tool URL anymore.   (On second thought, I could probably do this ahead of time in this PR...will fix shortly - thanks!)

hktest replied:
Fixed!

elliottyates commented on selenium_tests/course_admin/course_admin_base_test_case.py:
```diff
@@ -0,0 +1,31 @@
+
+        login_page = PinLoginPageObject(cls.driver)
+        if login_page.is_loaded():
+            login_page.login_xid(cls.USERNAME, cls.PASSWORD)
```
We're using XIDs for these tests?

hktest replied:
Yup, we've always been using an XID to login (for other projects as well), so I haven't changed the logic here.  We are using a mix of test data to test the Manage People tool, and that's reflected in the test data spreadsheet.  Let me know if there are new concerns here.

elliottyates commented:
👍

hktest committed:
Fixed data issue.  The successful HUID test were failing because the test ID is no findable in Harvard directory.

hktest committed:
- Removed all the get(self.base_url) that were built into the test.  No longer needed now that they're in setup.
- Unintentionally removed instantiating test_settings during setup cleanup.  Added back in.
- Verified that search tests runs/works.

hktest committed:
Removed the references to get URL for both add/remove tests - should be handled by setup.

hktest committed:
- Added a refresh to eliminate the element not found exception if the user is added via rest api but not yet appeared on the screen.  This was causing sporadically causing failures when running regression_suite.py

hktest committed:
Cleanup: no need to refer to the manage people URL, it's handled in setup.
Removed URL and comments, as they are no longer relevant for the tests.

hktest commented:
Tested and ran regression_suite.py successfully.  Merging into Elliott's branch.
